src/scala/com/github/johnynek/bazel_deps/MakeDeps.scala

```diff
@@ -88,16 +88,61 @@ object MakeDeps {
 
         // build the workspace
         val ws = Writer.workspace(g.depsFile, normalized, duplicates, shas, model)
-        g match {
-          case check: Command.CheckGen =>
-            executeCheckOnly(model, projectRoot, IO.path(check.shaFilePath), ws, targets, formatter)
-          case g: Command.Generate =>
-            // creates pom xml when path is provided
-            g.pomFile.foreach { fileName => CreatePom(normalized, fileName) }
-            executeGenerate(model, projectRoot, g.shaFilePath.map(IO.path(_)), g.targetFile.map(IO.path(_)), g.enable3rdPartyInRepo, ws, targets, formatter, g.resolvedOutput.map(IO.path),
-              artifacts)
+        // creates pom xml when path is provided
+        val pomIO = IO.orUnit(g.pomFile.map { fileName => CreatePom.writeIO(normalized, IO.path(fileName)) })
+        val io = executeGenerate(
+          model,
+          g.shaFilePath.map(IO.path(_)),
+          g.targetFile.map(IO.path(_)),
+          g.enable3rdPartyInRepo,
+          ws,
+          targets,
+          formatter,
+          g.resolvedOutput.map(IO.path),
+          artifacts) >> pomIO
+
+        if (g.checkOnly) {
```

Comment:
This is a pretty long `if {...}`. Maybe pull the content into a new function that returns an IO?